### DISCUSSION TITLE
Is it possible to prevent reducer and effect arguments from being typed as `any`.

### DISCUSSION THREAD

**Discussion by connebs** [rematch/rematch#963] [Q&A]
Basically title. Reducers and Effects in Rematch models have args automatically typed as any, which basically makes type checking when using them useless. Is there a way to turn this off so that I am forced to type all reducers and effects and get the guarantees that go along with that?

**Comment by tianzhich** [comment]
Could you give an example?

**Reply by connebs** [reply]
So in my project I have `@typescript-eslint/no-explicit-any: error` in ESLint and `noImplicitAny: true` in TSConfig, so if I have a function where args are not inferred or explicitly specified (and therefore are implicitly `any`), I want to know about it. *However*, Rematch is explicitly typing the args after the first `state` arg in a reducer as `any`, so my linter/compiler is not flagging that.

Example:

```ts
interface ModelState {
  val: boolean;
}

const model = createModel<RootModel>()({
  state: {
    val: true
  } as ModelState,
  reducers: {
    SET_VAL: (state, val) => ({ ...state, val })
  }
})

dispatch.model.SET_VAL("this should be boolean but is not caught because I forgot to type the reducer arg");
```

In the above `SET_VAL` reducer, my linter is not warning me that `val` is untyped because Rematch has set it explicitly to `any`, so I frequently forget to type values and then I lose type-checking benefits when calling that reducer via `dispatch.model.SET_VAL`. This is somewhat compounded by the fact that the type of the `state` arg is inferred for me as `ModelState` so makes it even easier to forget to type the 2nd arg.

**Reply by connebs** [reply]
Any thoughts?
